service/storage_service.cc

```diff
@@ -614,6 +607,9 @@ future<> storage_service::topology_state_load() {
             continue;
         }
         const auto ep = tmptr->get_endpoint_for_host_id(e);
```

Comment:
Yes. It looks like we skip something cdc related here.

Reply:
> > Yes. It looks like we skip something cdc related here.
> 
> `handle_cdc_generation` doesn't care about IPs, we don't need it in `gossiper_state_change_subscriber_proxy`.

But we skip it if the mapping is not available in topology load.

Reply:
> But we skip it if the mapping is not available in topology load.

I don't understand.

In `topology_state_load` we have
```
if (auto gen_id = _topology_state_machine._topology.current_cdc_generation_id) {
        slogger.debug("topology_state_load: current CDC generation ID: {}", *gen_id);
        co_await _cdc_gens.local().handle_cdc_generation(*gen_id);
    }
```
it doesn't need the mappings.

Reply:
Ah, right. It is outside of the loop, so it is not skipped.

Reply:
> there should be no regression since we wait for IPs before bootstrapping a node

Which you made optional (which I believe is incorrect) so the argument is invalid.

Reply:
made it non-optional